### DISCUSSION TITLE
Firefly not working on RaspberryPi 4

### DISCUSSION THREAD

**Discussion by david7867z** [firefly-iii/firefly-iii#4757] [Firefly III]
I am running Firefly III version 5.5.10.

Hello,
when i ran the docker image of firefly i got this error: "ERROR: no matching manifest for linux/arm/v7 in the manifest list entries"
like in issue #3464.
However if i changed the mysql image in the docker-compose to "arm64v8/mariadb:10.6" it succesfully created the containers, but it is still unreachable from the browser (i use port 8080 because pihole is on port 80 already).
I once got to the webpage and had an error 500.

**Comment by JC5** [comment]
Could you tell us what the 500 error was please?

**Reply by david7867z** [reply]
I think it was this one, i had it still in my clipboard:
Could not poll the database: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1)

**Reply by JC5** [reply]
Alright, remove the associated volumes and see if it runs when you restart it completely.

**Reply by david7867z** [reply]
I removed the volumes, restarted the pi and then used "docker-compose -f docker-compose.yml up -d" to start it.
I still cant reach the website, while pihole is reachable.
Also i just noticed, that only the "firefly_app_1"- container is up.

**Reply by JC5** [reply]
Not a surprise, looking closer. Your Raspberry PI is a ARM v7 but you download arm64**v8**/mariadb:10.6. The error from the database container will probably reflect this.

If you want a arm/v7 image (you do), try this image: https://hub.docker.com/r/yobasystems/alpine-mariadb/

**Reply by david7867z** [reply]
Ok, thanks i now changed the image i use, but i still cant reach the container. Both are now up.
When doing a cUrl (on the pi) for 192.168.178.6:8080 i get an "Connection refused"
when doing the same with localhost i get: "Connection reset by peer"

**Comment by david7867z** [comment]
![image](https://user-images.githubusercontent.com/54571318/117542088-6a5abd00-b017-11eb-865e-4efb0d6cfd8a.png)
This is the Error im getting now. For some reason i reached the site now.

**Reply by david7867z** [reply]
Also, the sql container exited, after restarting the message stayed the same.

**Reply by JC5** [reply]
I think that `DB_HOST` doesn't match the name of the database in the `docker-compose.yml` file

**Reply by david7867z** [reply]
The name of the database in the dockerfile is "MYSQL_DATABASE=firefly"
I didnt change anything except for the mysql version and the port

**Reply by david7867z** [reply]
DB_HOST is only in the .env

**Comment by david7867z** [comment]
What have you done, so it runs on your raspberry pi?

**Comment by RafaelFernandesBR** [comment]
Hello, I have the same problem, on the Raspberry Pi 3.

[+] Pulling 0/3                                                                      
 ⠴ db Pulling                                                                   2.6s 
 ⠴ cron 1 layers [⠀]      0B/0B      Pulling                                    2.6s 
 ⠴ app 24 layers [⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀]      0B/0B      Pulling             2.6s 
no matching manifest for linux/arm/v7 in the manifest list entries
